bridge/engine/EncodeJob.cpp

```diff
@@ -75,32 +77,31 @@ void EncodeJob::run()
         if (encodedBytes <= 0)
         {
             logger::error("Failed to encode opus, %d", "OpusEncodeJob", encodedBytes);
-            _outboundContext._allocator.free(_packet);
+            _audioPacketPoolAllocator.free(_packet);
             _packet = nullptr;
             return;
         }
 
-        std::memcpy(payloadStart, encodedData, encodedBytes);
-        _packet->setLength(headerLength + encodedBytes);
-        rtpPacket->ssrc = _outboundContext._ssrc;
-        rtpPacket->timestamp = (_rtpTimestamp * 48llu) & 0xFFFFFFFFllu;
-        rtpPacket->sequenceNumber = _outboundContext._sequenceCounter++ & 0xFFFFu;
-        rtpPacket->payloadType = targetFormat._payloadType;
-        doSend(_packet);
+        {
+            auto opusPacket = memory::makePacket(_outboundContext._allocator, rtpPacket, rtpPacket->headerLength());
+            auto opusHeader = rtp::RtpHeader::fromPacket(*opusPacket);
+            std::memcpy(opusHeader->getPayload(), encodedData, encodedBytes);
```

Comment:
I am wondering if we can't opus packet (`makePacket`) before call `opusEncoder->encode` and give opusHeader->getPayload() as encodedDataOutput, in the end is less one mem copy

Reply:
yes, we should be able to do that